app/Repositories/SignupRepository.php

```diff
@@ -43,6 +45,10 @@ public function create($data)
             $signup->save(['timestamps' => false]);
         }
 
+        // Now that the signup exists, set the signup_id on the signup event
```

Comment:
Ah this is getting complicated now. We first create the event, then the signup, the add the signup id to the event.  I wonder if this is something we can refactor later. Just noting it.